pkg/distributor/distributor.go

```diff
@@ -765,6 +772,18 @@ func (d *Distributor) Push(ctx context.Context, req *cortexpb.WriteRequest) (*co
 	d.receivedExemplars.WithLabelValues(userID).Add(float64(validatedExemplars))
 	d.receivedMetadata.WithLabelValues(userID).Add(float64(len(validatedMetadata)))
 
+	nhRateLimited := false
+	if limits.NativeHistogramsIngestionRate != math.MaxFloat64 {
```

Comment:
Just a nit. Maybe we can just call `d.nativeHistogramsIngestionRateLimiter.AllowN` and skip this check. I understand that we want to avoid calling limiter if it is max float but that should be similar to call AllowN with infinity strategy

Reply:
Thanks updated.